bitbots_vision/src/bitbots_vision/vision.py

```diff
@@ -86,17 +93,50 @@ def __init__(self):
         # Register VisionConfig server (dynamic reconfigure) and set callback
         srv = Server(VisionConfig, self._dynamic_reconfigure_callback)
 
-        rospy.spin()
+        # Run the vision main loop
+        self.main_loop()
+
+    def main_loop(self):
+        """
+        Main loop that processed the images and configuration changes
+        """
+        while not rospy.is_shutdown():
+            # Lookup if there is another configuration available
+            if self.transfer_reconfigure_data is not None:
+                # Copy config from shard memory
+                self.transfer_reconfigure_data_read_flag = True
+                reconfigure_data = deepcopy(self.transfer_reconfigure_data)
+                self.transfer_reconfigure_data_read_flag = False
+                self.transfer_reconfigure_data = None
+                # Run vision reconfiguration
+                self.configure_vision(*reconfigure_data)
+            # Check if a new image is avalabile
+            elif self.transfer_image_msg is not None:
+                # Copy image from shard memory
+                image_msg = deepcopy(self.transfer_image_msg)
+                self.transfer_image_msg = None
+                # Run the vision pipeline
+                self.handle_image(image_msg)
+                # Now the first image has been processed
+                self._first_image_callback = False
```

Comment:
do we need a sleep here i the else part?

Reply:
You mean, that the loop should not spin at insanely high rates if there's nothing to? We cloud add it, although I didn't noticed any weird behavior in the testing.